src/api/uploads/FolderUploadNode.js

```diff
@@ -96,34 +106,58 @@ class FolderUploadNode {
     createAndUploadFolder = async (errorCallback: Function, isRoot: boolean) => {
         await this.buildCurrentFolderFromEntry();
 
+        let errorEncountered = false;
+        let errorCode = '';
         try {
             const data = await this.createFolder();
             this.folderId = data.id;
         } catch (error) {
+            errorEncountered = true;
+            errorCode = error.code;
             // @TODO: Handle 429
             if (error.code !== ERROR_CODE_ITEM_NAME_IN_USE) {
-                errorCallback(error);
-                return;
+                if (!isRoot) {
+                    // If this is a child folder of the folder being uploaded, this errorCallback will set
+                    // an error message on the root folder being uploaded. Set a generic messages saying that a
+                    // child has caused the error. The child folder will be tagged with the error message in
+                    // the call to this.addFolderToUploadQueue below
+                    errorCallback({ code: ERROR_CODE_CHILD_FOLDER_FAILED_UPLOAD });
+                } else {
+                    errorCallback(error);
+                }
+            } else {
+                this.folderId = error.context_info.conflicts[0].id;
             }
-
-            this.folderId = error.context_info.conflicts[0].id;
         }
 
         // The root folder has already been added to the upload queue in ContentUploader
         if (isRoot) {
             return;
         }
 
-        this.addFolderToUploadQueue([
-            {
-                extension: '',
-                name: this.name,
-                status: STATUS_COMPLETE,
-                isFolder: true,
-                size: 1,
-                progress: 100,
-            },
-        ]);
+        const folderObject: {
+            extension: string,
+            name: string,
+            status: string,
+            isFolder: boolean,
+            size: number,
+            progress: number,
+            error?: Object,
+        } = {
+            extension: '',
+            name: this.name,
+            status: STATUS_COMPLETE,
+            isFolder: true,
+            size: 1,
+            progress: 100,
+        };
+
+        if (errorEncountered) {
+            folderObject.status = STATUS_ERROR;
+            folderObject.error = { code: errorCode };
+        }
+
+        this.addFolderToUploadQueue([folderObject]);
```

Comment:
I could not find a function with that name that takes array of objects?!?

Reply:
So it seems there is some wierd function naming going on. This points to ContentUploader.js::addToQueue() which takes array of UploadItem, which has error object

Reply:
Resolved, no longer calls with array.